src/App.jsx

```diff
@@ -63,17 +78,62 @@ function App() {
   //return [todoList, setTodoList];
   
 
-  function addTodo(newTodo){
+  const addTodo = async (newTodo) => {
     // add new items and keep old items too
-    setTodoList([...todoList, newTodo]);
-  }
+    const options = {
+      method: 'POST',
+      headers: {
+        'Authorization': `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`,
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        fields: {
+          title: newTodo.title,
+        },
+      }),
+    };
+    const url =
+      `https://api.airtable.com/v0/${import.meta.env.VITE_AIRTABLE_BASE_ID}/${import.meta.env.VITE_TABLE_NAME}`;
 
-  function removeTodo(id){
-    // get todo from todoList array
-    const newTodoList = todoList.filter(todo => todo.id !== id);
-    setTodoList(newTodoList);
-  }
+      try {
+        const response = await fetch(url, options);
+        if (!response.ok) {
+          const message = `Error: ${response.status}`;
+          throw new Error(message);
+        }
+        const data = await response.json();
+  
+        // Adding the new todo to state after adding to airtable
+        const addedTodo = {
+          id: data.id,
+          title: data.fields.title,
+        };
+  
+        setTodoList((prevTodos) => [...prevTodos, addedTodo]);
+      } catch (error) {
+        console.log('Error adding todo:', error.message);
+      }
+    };
+  
 
+  const removeTodo = async(id) => {
+    const options = {
+      method: 'DELETE',
+      headers: {
+        'Authorization': `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`,
+      },
+    };
+    const url = `https://api.airtable.com/v0/${import.meta.env.VITE_AIRTABLE_BASE_ID}/${import.meta.env.VITE_TABLE_NAME}/${id}`;
```

Comment:
Since you're constructing the URL more than once and it's pretty long, it may be a good idea to refactor so your component stays DRY. No need to make this change for this PR - just something for consideration.